File: My-Solutions/house-robber.java
class Solution {
    public int rob(int[] nums) {
        for(int i = 0; i<nums.length; i++){
            int cur_max = 0;
            for(int j = 0; j < i-1;j++){
                if(nums[j] > cur_max)
                    cur_max = nums[j];
            }
            nums[i] = cur_max + nums[i];
            System.out.println(nums[i]+":i == "+i);
        }
        
        int max = 0;
        
        for(int i = 0; i<nums.length; i++){
            if(nums[i] > max)
                max = nums[i];
        }
        return max;
    }
}

File: My-Solutions/house-robber.py
class Solution:
    def rob(self, nums: 'List[int]') -> 'int':
        if len(nums)==0:
            return 0
        dp = [0]*(len(nums)+2)
        dp[2] = nums[0]
        k = 3
        for i in range(1,len(nums)):
            dp[k] = max(dp[k-2],dp[k-3])+nums[i]
            k+=1
        return max(dp)
